cloud/pkg/devicecontroller/controller/downstream.go

```diff
@@ -167,6 +175,119 @@ func (dc *DownstreamController) syncDevice() {
 	}
 }
 
+// addDeviceMapper is function to add deviceMapper deployment
+func (dc *DownstreamController) addDeviceMapper(device *v1alpha2.Device, nodename string) {
+	protocol, err := getProtocolNameOfDevice(device)
+	if err != nil {
+		klog.Warningf("fail to get the protocol name of device %s with err: %v", device.Name, err)
+		return
+	}
+
+	mapperName := fmt.Sprintf("mapper-%s-%s", nodename, protocol)
+	_, err = dc.kubeClient.AppsV1().Deployments(device.Namespace).Get(context.Background(), mapperName, metav1.GetOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		klog.Warningf("Failed to get mapper deployment %s in namespace %s, error %v", mapperName, device.Namespace, err)
+		return
+	}
+
+	if err == nil {
+		klog.Infof("the mapper deployment %s has existed on the node %s. do not need to create one.", mapperName, nodename)
+		return
+	}
+
+	klog.Infof("the mapper deployment %s has not existed on the node %s. create one.", mapperName, nodename)
+
+	image, err := getImageOfDevice(device)
+	if err != nil {
+		klog.Warningf("fail to get the image of device mapper %s with err: %v", device.Name, err)
+		return
+	}
+
+	mdp := generateMapperDeployment(mapperName, device.Namespace, nodename, image)
+	if _, err := dc.kubeClient.AppsV1().Deployments(device.Namespace).Create(context.Background(), &mdp, metav1.CreateOptions{}); err != nil {
+		klog.Warningf("Failed to create mapper deployment %s in namespace %v, error %v", mapperName, device.Namespace, err)
+		return
+	}
+}
+
+// getProtocolNameOfDevice get the protocol name of a device
+func getProtocolNameOfDevice(device *v1alpha2.Device) (string, error) {
+	protocol := device.Spec.Protocol
+	if protocol.OpcUA != nil {
+		return OPCUA, nil
+	}
+	if protocol.Modbus != nil {
+		return Modbus, nil
+	}
+	if protocol.Bluetooth != nil {
+		return Bluetooth, nil
+	}
+	if protocol.CustomizedProtocol != nil {
+		return protocol.CustomizedProtocol.ProtocolName, nil
+	}
+	return "", fmt.Errorf("cannot find protocol name for device %s", device.Name)
+}
+
+// getImageOfDevice get the mapper image from the device annotation
+func getImageOfDevice(device *v1alpha2.Device) (string, error) {
+	if device.Annotations == nil {
+		return "", fmt.Errorf("device annotattion is empty")
+	}
+	image, ok := device.Annotations[Image]
+	if !ok {
+		return "", fmt.Errorf("device image is empty in annotattion")
+	}
+	return image, nil
+}
+
+// generateMapperDeployment generate the deployment for a mapper
+func generateMapperDeployment(mapperName, namespace, nodeName, mapperImage string) appv1.Deployment {
+	appLabelInfo := map[string]string{
+		App: mapperName,
+	}
+	nodeSelectorList := make(map[string]string)
+	nodeSelectorList[EdgeLabel] = ""
+	nodeSelectorList[NodeHostNameLabel] = nodeName
```

Comment:
I think we can not use this label to enable k8s to select the exact `nodeName` node to be scheduled?